### PR TITLE
TFLite: ConnectedComponents port from tf-addons.  Addition of  variable support.

### PR DESCRIPTION
**Pull request by bensternlieb**
First custom component.   Added a (sparse) BFLY.md readme as well.

### CODE REVIEW

File: BFLY.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+by the DL team for iOS model support. See the TF2 online
+documentation for bazel related targets and build info.
+
+In particular have some TFLite kernel code ported from
+the tf-addons repo here and some op registration
+for stateful variables.
+
+The principal iOS target here would be
+
+    //tensorflo/lite/ios:TensorFlowLiteC_framework
```
Comment (AlonDaks) [inline BFLY.md]:
```suggestion
    //tensorflow/lite/ios:TensorFlowLiteC_framework
```

File: tensorflow/lite/kernels/connected_components.cc [hunk 2]
```diff
@@ -0,0 +1,428 @@
+// This custom op is a port of the connected_components module from tensorflow_addons.
```
Comment (AlonDaks) [inline tensorflow/lite/kernels/connected_components.cc:1]: Can we also link to the reference, non tf-lite implementation?

File: tensorflow/lite/kernels/connected_components.cc [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+      connected_components::Free,
+      connected_components::Prepare,
+      connected_components::Eval
+  };
+  return &r;
+}
+
+}  // namespace custom
+}  // namespace ops
+}  // namespace tflite
```
Comment (AlonDaks) [inline tensorflow/lite/kernels/connected_components.cc]:
```suggestion
}  // namespace tflite

```

**Review by AlonDaks — APPROVED** [review]

File: tensorflow/lite/kernels/connected_components.cc (hunk 2 above)
Comment (bensternlieb) [reply tensorflow/lite/kernels/connected_components.cc:1]: Done!

**Pull request merged**